### ISSUE TITLE
Graceful handling of rate limiting?

### ISSUE THREAD

**Issue by DilumAluthge** [JuliaRegistries/RegistryCI.jl#424]
Right now, we just error as soon as we hit a rate limit. Instead, we should probably do this
1.	Ask GitHub when our rate limit will next reset. IIRC, we can do this with an API call that is not subject to rate limits.
2.	Compute the difference between the answer to #1 and the current time
3.	Sleep for the duration from #2
4.	Try the API call again

**Comment by oxinabox** [comment]
>  Ask GitHub when our rate limit will next reset. IIRC, we can do this with an API call that is not subject to rate limits.

Yes with this API https://docs.github.com/en/rest/reference/rate-limit#get-rate-limit-status-for-the-authenticated-user

I have a feeling it is also specified in the header of the returned error status
